File: CS/NavigationFrameWizard/WizardViewModel.cs
using DevExpress.Mvvm;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace NavigationFrameWizard {
    public class WizardViewModel : ViewModelBase{
        
        protected virtual ICurrentWindowService CurrentWindowService { get { return this.GetService<ICurrentWindowService>(); } }
        
        protected Customer _Customer;
        public Customer Customer {
            get {
                if(this._Customer == null) {
                    this._Customer = new Customer();
                }

                return this._Customer; 
            }
            set { this.SetProperty(ref this._Customer, value, "Customer"); }
        }


        protected bool? _DialogResult;
        public bool? DialogResult {
            get { return this._DialogResult; }
            set { this.SetProperty(ref this._DialogResult, value, "DialogResult"); }
        }

        #region Done

        protected DelegateCommand _Done;

        public DelegateCommand Done {
            get {
                if(this._Done == null) {
                    this._Done = new DelegateCommand(this.DoneExecute);
                }

                return this._Done;
            }
        }

        protected void DoneExecute() {
            this.DialogResult = true;
            this.CurrentWindowService.Close();
        }

        #endregion Done

        #region Cancel

        protected DelegateCommand _Cancel;

        public DelegateCommand Cancel {
            get {
                if(this._Cancel == null) {
                    this._Cancel = new DelegateCommand(this.CancelExecute);
                }

                return this._Cancel;
            }
        }

        protected void CancelExecute() {
            this.DialogResult = false;
            this.CurrentWindowService.Close();
        }

        #endregion Cancel
    }
}

File: VB/NavigationFrameWizard/WizardViewModel.vb
Imports DevExpress.Mvvm
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace NavigationFrameWizard
    Public Class WizardViewModel
        Inherits ViewModelBase


        Protected Overridable ReadOnly Property CurrentWindowService() As ICurrentWindowService
            Get
                Return Me.GetService(Of ICurrentWindowService)()
            End Get
        End Property

        Protected _Customer As Customer
        Public Property Customer() As Customer
            Get
                If Me._Customer Is Nothing Then
                    Me._Customer = New Customer()
                End If

                Return Me._Customer
            End Get
            Set(ByVal value As Customer)
                Me.SetProperty(Me._Customer, value, "Customer")
            End Set
        End Property


        Protected _DialogResult? As Boolean
        Public Property DialogResult() As Boolean?
            Get
                Return Me._DialogResult
            End Get
            Set(ByVal value? As Boolean)
                Me.SetProperty(Me._DialogResult, value, "DialogResult")
            End Set
        End Property

        #Region "Done"

        Protected _Done As DelegateCommand

        Public ReadOnly Property Done() As DelegateCommand
            Get
                If Me._Done Is Nothing Then
                    Me._Done = New DelegateCommand(AddressOf Me.DoneExecute)
                End If

                Return Me._Done
            End Get
        End Property

        Protected Sub DoneExecute()
            Me.DialogResult = True
            Me.CurrentWindowService.Close()
        End Sub

        #End Region ' Done

        #Region "Cancel"

        Protected _Cancel As DelegateCommand

        Public ReadOnly Property Cancel() As DelegateCommand
            Get
                If Me._Cancel Is Nothing Then
                    Me._Cancel = New DelegateCommand(AddressOf Me.CancelExecute)
                End If

                Return Me._Cancel
            End Get
        End Property

        Protected Sub CancelExecute()
            Me.DialogResult = False
            Me.CurrentWindowService.Close()
        End Sub

        #End Region ' Cancel
    End Class
End Namespace
